module-2/lab-matplotlib-seaborn/your-code/challenge-3.ipynb

```diff
@@ -9,7 +9,7 @@
   },
```

Comment:
- It's important to identify all categorical variables as categoric since the very begining. There is a special type of datatype called category that you can set with the astype method. An object type is not a categoric type. 
#your code here- categorical variable
titanic.select_dtypes(np.object).columns
- Now in addition to the summary statistics, we also want to visualize the distribution of Age. Choose the appropriate plot to sow below: you can also plot a violin plot to get both.
- We suspect that there is a linear relationship between Fare and Age. Use the appropriate plot to show the relationship between these 2 features. There are 2 ways, please do it both ways.
Hint: One of the ways involves using Seaborn: you can run a seaborn jointplot to get a scatter plot, an histogram and a KDE plots in the same. You can have all the information in just one place. 
```Python
sns.jointplot(x='Fare',y='Age',data=titanic)
```
- Good one!! i love correlations as a heatmap.However, I would adjust the color palette. Let's seat this afternoon to check how to play with color palettes in seaborn. 
```Python
ax = sns.heatmap(titanic.corr(), annot=True)
```
- Use the appropriate plot to display the distributions of  Age in relation to  Gender.
Hint: use Facetgrid to display the distribution of Age for each categorical value of Gender. It's better here to use histograms instead of violinplots. You can do it with a facetgrip and then a map method...
```Python
g = sns.FacetGrid(data=titanic,col='Gender')
g.map(plt.hist,'Age')
```